node/grpc/server_test.go

```diff
@@ -372,7 +410,79 @@ func TestAttestBatch(t *testing.T) {
 
 func TestRetrieveChunks(t *testing.T) {
 	server := newTestServer(t, true)
-	batchHeaderHash, _, _, _ := storeChunks(t, server)
+	batchHeaderHash, _, _, _ := storeChunks(t, server, false)
+
+	p := &peer.Peer{
+		Addr: &net.TCPAddr{
+			IP:   net.ParseIP("0.0.0.0"),
+			Port: 3000,
+		},
+	}
+	ctx := peer.NewContext(context.Background(), p)
+
+	retrievalReply, err := server.RetrieveChunks(ctx, &pb.RetrieveChunksRequest{
+		BatchHeaderHash: batchHeaderHash[:],
+		BlobIndex:       0,
+		QuorumId:        0,
+	})
+	assert.NoError(t, err)
+	assert.Equal(t, pb.ChunkEncoding_GOB, retrievalReply.Encoding)
+	recovered, err := new(encoding.Frame).Deserialize(retrievalReply.GetChunks()[0])
+	assert.NoError(t, err)
+	chunk, err := new(encoding.Frame).Deserialize(encodedChunk)
+	assert.NoError(t, err)
+	assert.Equal(t, recovered, chunk)
+
+	retrievalReply, err = server.RetrieveChunks(ctx, &pb.RetrieveChunksRequest{
+		BatchHeaderHash: batchHeaderHash[:],
+		BlobIndex:       0,
+		QuorumId:        1,
+	})
+	assert.NoError(t, err)
+	assert.Equal(t, pb.ChunkEncoding_UNKNOWN, retrievalReply.Encoding)
```

Comment:
nice, so you handled the case when bundle is empty when operator is not in that quorum. But you also return the unknown when blob cannot be retrieved, but since you return the error there,so it would fine I think

Reply:
It should be fine, empty bundle has no header to tell what's the encoding format anyway.